openpype/hosts/houdini/api/lib.py

```diff
@@ -1020,21 +1070,38 @@ def self_publish():
         return
 
     current_node = hou.node(".")
-    inputs_paths = find_rop_input_dependencies(
-        current_node.inputDependencies()
-    )
-    inputs_paths.append(current_node.path())
+    all_input_nodes = find_rop_inputs_chain(current_node)
+    all_input_nodes.append(current_node)
+
+    all_input_subsets = [n.evalParm("subset") for n in all_input_nodes]
 
     host = registered_host()
     context = CreateContext(host, reset=True)
 
+    undesired_instances = []
     for instance in context.instances:
-        node_path = instance.data.get("instance_node")
-        instance["active"] = node_path and node_path in inputs_paths
+        if instance.get("subset") not in all_input_subsets:
+            undesired_instances.append(instance)
+            instance["active"] = False
+            continue
+
+        # make sure the instance is active
+        instance["active"] = True
 
     context.save_changes()
 
-    publisher_show_and_publish(comment)
+    publish_mode_parm = current_node.parm("ayon_self_publish_selection")
+    if publish_mode_parm and publish_mode_parm.eval():
+        publisher_show_and_publish(comment)
+        return
+
+    for instance in undesired_instances:
+        log.debug("Remove {} instance".format(instance.get("subset")))
+        context.creator_removed_instance(instance)
```

Comment:
This seems overly 'aggressive'. It should be sufficient to just disable the instances, right? Instead of forcefully removing them?

I think the instance's data might also need to have `instance.data["publish"] = False` since that was the case in this Substance Painter PR here: https://github.com/ynput/OpenPype/pull/6127

Reply:
oh, I didn't know that disabling the instances is enough! 
I've tested it.. going to push my commit.